src/fragments/lib/graphqlapi/js/optimistic-ui.mdx

```diff
@@ -374,17 +375,22 @@ const updateMutation = useMutation({
   onError: (err, newRealEstateProperty, context) => {
     console.error("Error updating record:", err, newRealEstateProperty);
     if (context?.previousRealEstateProperty) {
-      queryClient.setQueryData<RealEstateProperty>(
+      queryClient.setQueryData(
         ["realEstateProperties", context.newRealEstateProperty.id],
         context.previousRealEstateProperty
       );
     }
   },
   // Always refetch after error or success:
   onSettled: (newRealEstateProperty) => {
-    queryClient.invalidateQueries({
-      queryKey: ["realEstateProperties", newRealEstateProperty.id],
-    });
+    if (newRealEstateProperty) {
+      queryClient.invalidateQueries({
+        queryKey: ["realEstateProperties", newRealEstateProperty.id],
+      });
+      queryClient.invalidateQueries({
```

Comment:
is this case covered in 387?

Reply:
Both invalidations here are required - one invalidates the cache for the list view, the other invalidates the cache for the detail view.